dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+        }
+
+        for (File file : contains("com.dotmarketing.portlets.workflows.model.WorkflowTask_")) {
+            doXMLFileImport(file);
+        }
+        for (File file : contains("com.dotmarketing.portlets.workflows.model.WorkflowHistory_")) {
+            doXMLFileImport(file);
+        }
+        for (File file : contains("com.dotmarketing.portlets.workflows.model.WorkflowComment_")) {
+            doXMLFileImport(file);
+        }
+        for (File file : contains("com.dotmarketing.portlets.workflows.model.WorkFlowTaskFiles_")) {
+            doXMLFileImport(file);
+        }
+
+        for (File file : contains("com.dotmarketing.tag.model.Tag_")) {
+            doXMLFileImport(file);
+        }
+        // Image, Portlet, Multitree, category
+        for (File file : endsWith(".xml")) {
+            doXMLFileImport(file);
+        }
+
+
+        cleanUpDBFromImport();
+        Optional<File> assetDir = tempFiles.stream().filter(f -> ("asset".equals(f.getName())||"assets".equals(f.getName())) && f.isDirectory()).findAny();
+        if (assetDir.isPresent()) {
+            copyAssetDir(assetDir.get());
+        }
+        Logger.info(ImportStarterUtil.class, "Done Importing");
+        deleteTempFiles();
+
+
+
+    }
+
+    /**
+     *
+     * @param fromAssetDir
+     * @throws IOException
+     */
+    private void copyAssetDir(File fromAssetDir) throws IOException {
+        File ad = new File(assetPath);
+
+        ad.mkdirs();
+        String[] fileNames = fromAssetDir.list();
+        for (int i = 0; i < fileNames.length; i++) {
+            File f = new File(fromAssetDir.getPath() + File.separator + fileNames[i]);
+            if (f.getName().equals(".svn")) {
+                continue;
+            }
+            if (f.getName().equals("license.dat")) {
+                continue;
+            }
+            if (f.isDirectory()) {
+                FileUtil.copyDirectory(f.getPath(), ad.getPath() + File.separator + f.getName());
+            } else {
+                FileUtil.copyFile(f.getPath(), ad.getPath() + File.separator + f.getName());
+            }
+        }
+    }
+
+    /**
+     * Deletes all files from the backupTempFilePath
+     */
+    private void deleteTempFiles() {
+        File f = new File(backupTempFilePath);
+
+        FileUtil.deltree(f, true);
+    }
+
+
+
+    private void deleteTable(final String table) {
+
+        final DotConnect dotConnect = new DotConnect();
+        dotConnect.setSQL("delete from " + table);
+        dotConnect.getResult();
+    }
+
+    interface ObjectFilter {
+        boolean includeIt(Object obj);
+    }
+
+    /**
+     * This method takes an XML file and will try to import it via XStream and Hibernate.
+     *
+     * @param f - File to be parsed and imported.
+     * @param out - Printwriter to write responses to Reponse Printwriter so this method can write to
+     *        screen.
+     */
+    private void doXMLFileImport(File f) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'f' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)